### DISCUSSION TITLE
Out of sync schema in LB scenario

### DISCUSSION THREAD

**Discussion by nichols-green** [directus/directus#11851] [Archive]
We host our Directus site on two separate VMs, and load balance the two nodes. Whenever we make a schema change through the Directus UI, the other node is not aware of the changes and is out of sync until we restart the service. We are using SQL Server. Does anyone know of a way to keep the two nodes synchronized?

**Comment by rijkvanzanten** [comment]
Yes, use a shared storage for caching, like Redis or Memcached 🙂 The out-of-sync behavior you're seeing is from the schema cache not being purged properly on instance B when instance A is making changes. Having them both rely on a shared caching storage prevents these types of projects. While we're on the subject, make sure to do the same for file storage, by relying on a non-local storage (unless both node processes can access the same FS)

**Reply by nichols-green** [reply]
Thanks for the fast feedback, I'll work on getting a cache implemented.

**Reply by rijkvanzanten** [reply]
In the meantime, you could also disable caches completely (see the various flags in https://docs.directus.io/configuration/config-options/#cache), though I wouldn't recommend that approach for production use, as it does hinder performance 👍🏻
